cmd/kpod/push.go

```diff
@@ -0,0 +1,209 @@
+	// specified, indicating that the shared, system-wide default policy
+	// should be used.
+	SignaturePolicyPath string
+	// ReportWriter is an io.Writer which will be used to log the writing
+	// of the new image.
+	ReportWriter io.Writer
+	// Store is the local storage store which holds the source image.
+	Store storage.Store
+	// DockerRegistryOptions encapsulates settings that affect how we
+	// connect or authenticate to a remote registry to which we want to
+	// push the image.
+	dockerRegistryOptions
+	// SigningOptions encapsulates settings that control whether or not we
+	// strip or add signatures to the image when pushing (uploading) the
+	// image to a registry.
+	signingOptions
+}
+
+func pushCmd(c *cli.Context) error {
+	var registryCreds *types.DockerAuthConfig
+
+	args := c.Args()
+	if len(args) < 2 {
+		return errors.New("kpod push requires exactly 2 arguments")
+	}
+	srcName := c.Args().Get(0)
+	destName := c.Args().Get(1)
+
+	signaturePolicy := ""
+	if c.IsSet("signature-policy") {
+		signaturePolicy = c.String("signature-policy")
+	}
+	compress := archive.Uncompressed
+	if !c.IsSet("disable-compression") || !c.Bool("disable-compression") {
+		compress = archive.Gzip
+	}
+	quiet := false
+	if c.IsSet("quiet") {
+		quiet = c.Bool("quiet")
+	}
+	registryCredsString := ""
+	if c.IsSet("creds") {
+		registryCredsString = c.String("creds")
+	}
+	certPath := ""
+	if c.IsSet("cert-dir") {
+		certPath = c.String("cert-dir")
+	}
+	skipVerify := false
+	if c.IsSet("tls-verify") {
+		skipVerify = !c.BoolT("tls-verify")
+	}
+	removeSignatures := false
+	if c.IsSet("remove-signatures") {
+		removeSignatures = c.Bool("remove-signatures")
+	}
+	signBy := ""
+	if c.IsSet("sign-by") {
+		signBy = c.String("sign-by")
+	}
+
+	if registryCredsString != "" {
+		creds, err := parseRegistryCreds(registryCredsString)
+		if err != nil {
+			return err
+		}
+		registryCreds = creds
+	}
+
+	store, err := getStore(c)
+	if err != nil {
+		return err
+	}
+
+	options := pushOptions{
+		Compression:         compress,
+		SignaturePolicyPath: signaturePolicy,
+		Store:               store,
+		dockerRegistryOptions: dockerRegistryOptions{
+			DockerRegistryCreds:         registryCreds,
+			DockerCertPath:              certPath,
+			DockerInsecureSkipTLSVerify: skipVerify,
+		},
+		signingOptions: signingOptions{
+			RemoveSignatures: removeSignatures,
+			SignBy:           signBy,
+		},
+	}
+	if !quiet {
+		options.ReportWriter = os.Stderr
+	}
+	return pushImage(srcName, destName, options)
+}
+
+func pushImage(srcName, destName string, options pushOptions) error {
+	if srcName == "" || destName == "" {
+		return errors.Errorf("source and destination image names must be specified")
+	}
+
+	// Get the destination Image Reference
+	dest, err := alltransports.ParseImageName(destName)
+	if err != nil {
+		return errors.Wrapf(err, "error getting destination imageReference for %q", destName)
+	}
+
+	policyContext, err := getPolicyContext()
```

Comment:
`policyContext.Destroy() ` should eventually be called. Right now that does not matter much, but eventually the context should be carrying cached state and failing to call this could cause files and processes to be left around.